pkg/controller/container-runtime-config/helpers.go

```diff
@@ -296,3 +297,29 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
 
 	return nil
 }
+
+// getValidRegistries gets the insecure and blocked registries in the image spec and validates that the user is not adding
+// the registry being used by the payload to the list of blocked registries.
+// If the user is, we drop that registry and continue with syncing the registries.conf with the other registry options
+func getValidRegistries(clusterVersionStatus *apicfgv1.ClusterVersionStatus, imgSpec *apicfgv1.ImageSpec) ([]string, []string, error) {
+	if clusterVersionStatus == nil || imgSpec == nil {
+		return nil, nil, nil
+	}
+
+	var blockedRegs []string
+	// Copy the insecure registries from the spec
+	insecureRegs := imgSpec.RegistrySources.InsecureRegistries
+
+	// Get the registry being used by the payload from the clusterversion config
+	payloadReg := strings.Split(clusterVersionStatus.Desired.Image, "/")[0]
```

Comment:
This looks fragile, can you use something from c/image to split domain components?

Reply:
fixed